clients/src/main/resources/common/message/CreateTopicsRequest.json

```diff
@@ -53,7 +53,9 @@
           "about": "The configuration name." },
         { "name": "Value", "type": "string", "versions": "0+", "nullableVersions": "0+",
           "about": "The configuration value." }
-      ]}
+      ]},
+      { "name": "Id", "type": "uuid", "tag": 0, "taggedVersions":  "5+", "versions": "5+", "ignorable":  true,
+        "about": "Optional topic id."}
```

Comment:
issue: We need to consider what happens if upstream adds a tagged field to this model. Would there be a graceful upgrade path in that case?

As an alternative, would it be feasible to use an entirely separate API, by copying this schema and somehow mapping it to the same internal endpoint? API key [is a signed field](https://github.com/apache/kafka/blob/60e845626d8a465a8cfe68bb2d7d4b88d622634e/clients/src/main/resources/common/message/RequestHeader.json#L27) so we could potentially use a negative value.

Reply:
> We need to consider what happens if upstream adds a tagged field to this model. Would there be a graceful upgrade path in that case?

We would then need to update the API to increment the tag of the `Id` field. I don't think there's any other way around.

> would it be feasible to use an entirely separate API, by copying this schema and somehow mapping it to the same internal endpoint?

I tried to implement it, but unfortunately this feature would then impact many more parts of the codebase. I think we cannot go through this route.

Another option would be to just leave the APIs intact, and instead add the topic id as a special config that will be interpreted as the topic id instead of a normal topic config. In this way we would then be safe from all possible future API changes.

On the other side, `CreateTopics` didn't receive any update since 2021, when topic id were introduced, and I don't seem to find any relevant KIP that would change the API in the near future.